_posts/2020-07-16-building-drivers-on-zephyr.md

```diff
@@ -0,0 +1,644 @@
+---
+title: "Tips on Building Drivers for Zephyr"
+description: "A guide on how to build custom drivers for hardware on Zephyr."
+author: jaredwolff
+---
+
+Over the past couple weeks i've been getting to know the insides and outsides of Zephyr. Since coming from Nordic Semiconductor's bare metal SDK, it's been an interesting experience.
+
+<!-- excerpt start -->
+In this post, i'll go over some of the nuances related to creating drivers for your peripherals on Zephyr. We'll talk about Device Tree, organization and more. By the end you should have an idea how to tackle your own projects!
+<!-- excerpt end -->
+
+## Table of Contents
+
+<!-- prettier-ignore -->
+* auto-gen TOC:
+{:toc}
+
+## Setup + Tools
+
+There are some prerequisites to getting started. Let's begin with SDK Setup.
+
+### SDK Setup
+
+I'll be referencing Nordic's nRF Connect SDK a bunch throughout this post. I've been building and testing nRF9160 Feather on v1.3.0. I won't be going into detail on how to set up the SDK though. Nordic has [great documentation](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/getting_started.html) you can check out to get started. I've also previously discussed [my experience here](https://www.jaredwolff.com/zephyr-on-nrf9160-feather/).
+
+### What `west` can do for you
+
+`west` is the de facto tool for Zephyr projects. It makes project management, compilation and flashing easier. It was built to replace the idea of a top level Makefile and project dependency management like Git submodules. For the most part, it accomplishes that goal.
+
+You'll definitely need to install `west` if you haven't already. Follow the SDK setup links above for more information.
+
+**Building**
+
+To build a project you'll either have to change directories to the application in question or provide the path. For example:
+
+```terminal
+cd ncs/nrf/samples/nrf9160/nrf_cloud_agps
```

Comment:
nit: prefix shell commands with $ for readability

Reply:
Adjusted